shuup/gdpr/receivers.py

```diff
@@ -31,4 +33,8 @@ def create_consents_login_allowed(sender, request, user, *args, **kwargs):
 
 @receiver(checkout_complete)
 def create_consents_checkout_complete(sender, request, user, order, *args, **kwargs):
-    create_user_consent_for_all_documents(request.shop, user)
+    gdpr_settings = GDPRSettings.get_for_shop(request.shop)
+    if not user.is_authenticated() and has_installed("shuup.gdpr") and gdpr_settings.enabled:
+        create_user_consent_for_all_documents(request.shop, order.orderer.user)
```

Comment:
Maybe `order.creator` since not sure someone can add consent behalf of someone else since here order can be created as staff as well.